level-2/index.html

```diff
@@ -148,10 +148,15 @@ <h4>
             <li>Let <var>p</var> be <a data-cite=
             "!promises-guide#a-new-promise">a new promise</a>.
             </li>
-            <li>If none of <var>data</var>'s members <a>title</a>, <a>text</a>,
-            <a>url</a> or (if the implementation supports file sharing)
-            <a>files</a> are present, reject <var>p</var> with
-            <a>TypeError</a>, and abort these steps.
+            <li>If none of <var>data</var>'s members <a>title</a>, <a>text</a>
+            or <a>url</a> are present:
+              <ol>
+                <li>If <var>data</var>'s member <a>files</a> is not present or
+                is empty, or if the implementation does not support file
+                sharing, reject <var>p</var> with <a>TypeError</a>, and abort
```

Comment:
I'm wondering whether it would be better to use a different type of error if file sharing isn't supported? This preserves the existing behavior though, so seems fine for now.

Reply:
Also, should the title of this PR be "Don't share if files is empty..."?

Reply:
Using TypeError is convenient here as canShare needs to return false, and canShare returns false whenever the share would fail with a TypeErrors.